compiler/x/codegen/X86Debug.cpp

```diff
@@ -239,31 +239,32 @@ TR_Debug::printDependencyConditions(
       cursor += 12;
 
       *(cursor++) = '(';
-      TR::RealRegister::RegNum r = (conditions->getRegisterDependency(i))->getRealRegister();
-      if  (r == TR::RealRegister::AllFPRegisters)
+      TR::RegisterDependency *regDep = conditions->getRegisterDependency(i);
+      if (regDep->isAllFPRegisters())
          {
          len = sprintf(cursor, "AllFP");
          }
-      else if (r == TR::RealRegister::NoReg)
+      else if (regDep->getRealRegister() == TR::RealRegister::NoReg)
```

Comment:
Should this be abstracted as well? I know on x86 and non-Z (?) platforms we use `NoReg`. On Z we use `AssignAny`. I'd like to get rid of this difference and abstracting this may be a step in that direction.

Reply:
In my first implementation I was on the fence for this one but decided to include it in the next round of pushes.